packages/components/tests/dummy/app/controllers/components/code-block.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { scheduleOnce } from '@ember/runloop';
+
+function replaceMockCopyStatus() {
+  document.querySelectorAll('[mock-copy-status]').forEach((element) => {
+    const status = element.attributes['mock-copy-status'].value;
+    element.classList.remove('hds-copy-button--status-idle');
+    element.classList.add(`hds-copy-button--status-${status}`);
+    const icon = element.querySelector('svg use');
+    if (icon) {
+      if (status === 'success') {
+        // eg. href="#flight-clipboard-checked-16"
+        icon.setAttribute('href', `#flight-${this.model.SUCCESS_ICON}-16`);
```

Comment:
@alex-ju @Dhaulagiri FYI there was an issue in the code that triggered a JS error here, when the user was navigating from the index page of the showcase to the `code-block` page (as well as the `copy/button` and `copy/snippet` pages. I have pushed a commit with the fix here: https://github.com/hashicorp/design-system/pull/1687/commits/b82b9f9716ff215933db702a81c5f8ddb2e3294a